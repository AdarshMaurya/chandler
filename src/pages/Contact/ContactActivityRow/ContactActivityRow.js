import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {Text, View, Image} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

import {Touchable} from 'components';
import {styles} from './ContactActivityRow.styles';
import {commonStyles} from 'theme';

export class ContactActivityRow extends Component {
  static propTypes = {
    count: PropTypes.number.isRequired,
    image: PropTypes.any.isRequired,
    title: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired,
    last: PropTypes.bool,
  };
  render() {
    const {count, image, title, last, onPress} = this.props;

    return (
      <View>
        <Touchable style={styles.container} onPress={onPress}>
          <Image source={image} style={styles.image} />
          <View style={commonStyles.flex}>
            <Text>{title.toUpperCase()}</Text>
          </View>
          <Text>{count}</Text>
          <Icon
            name="ios-arrow-forward"
            size={25}
            color="#AAA"
            style={styles.icon}
          />
        </Touchable>
        {!last && <View style={styles.border} />}
      </View>
    );
  }
}
