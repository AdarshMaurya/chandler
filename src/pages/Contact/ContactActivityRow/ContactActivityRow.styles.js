import {StyleSheet, PixelRatio} from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 50,
    alignItems: 'center',
  },
  image: {
    marginLeft: 5,
    marginRight: 10,
  },
  border: {
    height: 1 / PixelRatio.get(),
    backgroundColor: '#E2E4E5',
  },
  icon: {
    marginLeft: 10,
  },
});
