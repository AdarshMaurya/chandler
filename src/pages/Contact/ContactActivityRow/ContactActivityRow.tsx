import React, {PureComponent} from 'react';
import {Image, Text, View} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

import {Touchable} from '@components';
import {commonStyles} from '@theme';
import {styles} from './ContactActivityRow.styles';

interface IContactActivityRowProps {
  count: number;
  image: number | {uri: string};
  title: string;
  onPress: (...args: any[]) => any;
  last?: boolean;
}

export class ContactActivityRow extends PureComponent<
  IContactActivityRowProps,
  {}
> {
  public render() {
    const {count, image, title, last, onPress} = this.props;

    return (
      <View>
        <Touchable style={styles.container} onPress={onPress}>
          <Image source={image} style={styles.image} />
          <View style={commonStyles.flex}>
            <Text>{title.toUpperCase()}</Text>
          </View>
          <Text>{count}</Text>
          <Icon
            name="ios-arrow-forward"
            size={25}
            color="#AAA"
            style={styles.icon}
          />
        </Touchable>
        {!last && <View style={styles.border} />}
      </View>
    );
  }
}
