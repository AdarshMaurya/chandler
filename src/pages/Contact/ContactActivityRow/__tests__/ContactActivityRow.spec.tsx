import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import {ContactActivityRow} from '../ContactActivityRow';

describe('Pages', () => {
  describe('Login', () => {
    let onPress;

    beforeEach(() => {
      onPress = jest.fn();
    });

    it('should renders correctly', () => {
      expect(
        toJson(
          shallow(
            <ContactActivityRow
              count={0}
              image={{uri: 'image.net'}}
              title="My title"
              onPress={onPress}
            />
          )
        )
      ).toMatchSnapshot();
    });

    it('should renders correctly with last true', () => {
      expect(
        toJson(
          shallow(
            <ContactActivityRow
              count={0}
              image={{uri: 'image.net'}}
              title="My title"
              onPress={onPress}
              last={true}
            />
          )
        )
      ).toMatchSnapshot();
    });

    it('should call onPress when taping the row', () => {
      const row = shallow(
        <ContactActivityRow
          count={0}
          image={{uri: 'image.net'}}
          title="My title"
          onPress={onPress}
        />
      );
      row.find('Touchable').prop('onPress')();
      expect(onPress).toHaveBeenCalledTimes(1);
      expect(onPress).toHaveBeenCalledWith();
    });
  });
});
