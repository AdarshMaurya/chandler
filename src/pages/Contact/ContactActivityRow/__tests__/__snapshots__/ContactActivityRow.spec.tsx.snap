// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Login should renders correctly 1`] = `
<Component>
  <Touchable
    onPress={[MockFunction]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "image.net",
        }
      }
      style={
        Object {
          "marginLeft": 5,
          "marginRight": 10,
        }
      }
    />
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        MY TITLE
      </Text>
    </Component>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      0
    </Text>
    <Icon
      allowFontScaling={false}
      color="#AAA"
      name="ios-arrow-forward"
      size={25}
      style={
        Object {
          "marginLeft": 10,
        }
      }
    />
  </Touchable>
  <Component
    style={
      Object {
        "backgroundColor": "#E2E4E5",
        "height": 0.5,
      }
    }
  />
</Component>
`;

exports[`Pages Login should renders correctly with last true 1`] = `
<Component>
  <Touchable
    onPress={[MockFunction]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "image.net",
        }
      }
      style={
        Object {
          "marginLeft": 5,
          "marginRight": 10,
        }
      }
    />
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        MY TITLE
      </Text>
    </Component>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      0
    </Text>
    <Icon
      allowFontScaling={false}
      color="#AAA"
      name="ios-arrow-forward"
      size={25}
      style={
        Object {
          "marginLeft": 10,
        }
      }
    />
  </Touchable>
</Component>
`;
